### PR TITLE
Close Air Support TLC pack

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
- Cleans up CAS code a bit
- Makes debugging cas code MUCH easier. Use TESTING to be able to instantly take the Condor planetside.
- Adds two lase features
## Changelog
:cl:
add: CAS will now get JUMP links in chat when a lase is deployed.
add: CAS now has a Jump to Lase ability button.
qol: Clicking the Condor cockpit as the pilot will now attempt to move you out. No more resist needed.
code: Cleaned up CAS code. Please contribute CAS features. I beg you.
/:cl:

### CODE REVIEW

File: code/modules/condor/cas_misc.dm
Comment (TiviPlus) [inline code/modules/condor/cas_misc.dm:1]: Please dont use misc files for love of god just name after whats in em (actions)

File: code/modules/condor/cas_chair.dm [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+			if(PLANE_STATE_PREPARED)
+				owner.turn_off_engines()
+
+	if(owner.state == PLANE_STATE_ACTIVATED)
+		return
+
+	switch(action)
+		if("launch")
+			if(!cas_usable)
+				to_chat(usr, "<span class='warning'>Combat has not yet initiated, CAS unavailable.")
```
Comment (TiviPlus) [inline code/modules/condor/cas_chair.dm:205]: use spans

File: code/modules/condor/cas_chair.dm [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+				return
+			if(owner.state == PLANE_STATE_FLYING || owner.mode != SHUTTLE_IDLE)
+				return
+			if(owner.fuel_left <= LOW_FUEL_THRESHOLD)
+				to_chat(usr, "<span class='warning'>Unable to launch, low fuel.")
```
Comment (TiviPlus) [inline code/modules/condor/cas_chair.dm:210]: ditto, or even better use balloon alerts

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Please split your File move and changes in future or I will force you to atomize

File: code/__DEFINES/cas.dm [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#define LOW_FUEL_LANDING_THRESHOLD 4
+
+#ifdef TESTING
+#define CAS_USABLE TRUE
+#else
+#define CAS_USABLE FALSE
+#endif
+
+///Provide a link for CAS users to click to jump to. Pair with /proc/message_cas()
+#define CAS_JUMP_LINK(location) "(<a href='?_src_=usr;cas_jump=\ref[location];'>JUMP</a>)"
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/__DEFINES/cas.dm:18]: It was originally but I later redid how I handled the JUMP message. I wanted to keep it in to incentivize people to make more CAS features.

File: code/modules/condor/cas_misc.dm [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/// Send a to_chat() to all ongoing CAS operations.
+/proc/message_cas(text)
+	for(var/obj/docking_port/mobile/marine_dropship/casplane/plane in GLOB.active_cas_planes)
+		to_chat(plane.eyeobj.eye_user, text)
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/condor/cas_misc.dm:4]: See previous comment.

File: code/modules/condor/cas_chair.dm (hunk 2 above)
Comment (Kapu1178) [reply code/modules/condor/cas_chair.dm:210]: Leftover from file move.

File: code/modules/condor/cas_chair.dm [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+	RegisterSignal(SSdcs, COMSIG_GLOB_CAS_LASER_CREATED, PROC_REF(receive_laser_cas))
+	RegisterSignal(SSdcs, list(COMSIG_GLOB_OPEN_TIMED_SHUTTERS_LATE, COMSIG_GLOB_OPEN_TIMED_SHUTTERS_XENO_HIVEMIND, COMSIG_GLOB_OPEN_SHUTTERS_EARLY, COMSIG_GLOB_TADPOLE_LAUNCHED), PROC_REF(cas_usable))
+
+/obj/structure/caspart/caschair/Destroy()
+	owner?.chair = null
+	owner = null
+	UnregisterSignal(SSdcs, COMSIG_GLOB_CAS_LASER_CREATED)
+	if(occupant)
+		INVOKE_ASYNC(src, PROC_REF(eject_user), TRUE)
+	QDEL_NULL(cockpit)
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/condor/cas_chair.dm:32]: Moved file contents, did not author.

File: code/modules/condor/cas_shuttle.dm [hunk 6]
```diff
@@ -213,6 +53,7 @@
 	return ..()
 
 /obj/docking_port/mobile/marine_dropship/casplane/process()
+	#ifndef TESTING
 	fuel_left--
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/condor/cas_shuttle.dm:57]: I thought it'd be unlikely fuel mechanics would be changed soon (if ever), indefinite flight makes testing anything that happens during flight easier. Will revert if asked.

File: code/modules/condor/cas_chair.dm (hunk 1 above)
Comment (Kapu1178) [reply code/modules/condor/cas_chair.dm:205]: Moved file contents, did not author.

File: code/modules/condor/cas_shuttle.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/condor/cas_shuttle.dm:57]: mhmmm I guess

File: code/__DEFINES/cas.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/__DEFINES/cas.dm:18]: its dead code, and since CAS_JUMP_LINK (the hard part) exists it's kinda silly

**Pull request merged**